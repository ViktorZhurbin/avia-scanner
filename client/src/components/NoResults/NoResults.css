.container {
    width: 100%;
    max-width: var(--max-ticket-width);
    min-height: 140px;
    padding: 15px 25px;
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    border-radius: 5px;
    box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.15);
    background-color: var(--color-white);
    color: var(--color-twilight);
}

.text {
    font: var(--h5);
    margin: auto;
    margin-bottom: 0;
}

.resetButton {
    margin-top: 25px;
    cursor: pointer;
    user-select: none;
    max-width: 165px;
}

.buttonText {
    line-height: 45px;
}
