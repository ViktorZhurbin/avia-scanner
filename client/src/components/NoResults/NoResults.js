import React from 'react';
import PropTypes from 'prop-types';
import cl from 'classnames/bind';

import Button from '../Button';

import styles from './NoResults.css';

const cx = cl.bind(styles);

class NoResults extends React.PureComponent {
    static propTypes = {
        stopOptions: PropTypes.arrayOf(
            PropTypes.number,
        ).isRequired,
        setUpStops: PropTypes.func.isRequired,
        count: PropTypes.number.isRequired,
    }

    onFilterReset = () => {
        const { stopOptions, setUpStops } = this.props;
        const resetFilterState = {
            [stopOptions[0]]: true,
        };

        setUpStops(resetFilterState);
    }

    render() {
        const { count } = this.props;

        return (
            <div className={cx('container')}>
                <div className={cx('text')}>
                    {`We found ${count} tickets
            but none of them matches current filter settings`}
                </div>
                <div className={cx('resetButton')}>
                    <Button
                        onClick={this.onFilterReset}
                    >
                        <span className={cx('buttonText')}>
                            Reset Filters
                        </span>
                    </Button>
                </div>
            </div>
        );
    }
}

export default NoResults;
