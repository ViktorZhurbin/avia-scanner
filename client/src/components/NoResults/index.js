import React from 'react';
import PropTypes from 'prop-types';
import cl from 'classnames/bind';

import Button from '../Button';

import styles from './index.css';

const cx = cl.bind(styles);

const NoResuts = ({ ticketsCount, onFilterReset }) => (
    <div className={cx('container')}>
        <div className={cx('text')}>
            {`We found ${ticketsCount} tickets
            but none of them matches current filter settings`}
        </div>
        <div className={cx('resetButton')}>
            {/* eslint-disable-next-line */}
            <Button
                onClick={onFilterReset}
            >
                <span className={cx('buttonText')}>
                    Reset Filters
                </span>
            </Button>
        </div>
    </div>
);

NoResuts.propTypes = {
    ticketsCount: PropTypes.number.isRequired,
    onFilterReset: PropTypes.func.isRequired,
};

export default React.memo(NoResuts);
