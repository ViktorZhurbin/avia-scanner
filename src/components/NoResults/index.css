.container {
    width: 100%;
    min-width: 611px;
    min-height: 140px;
    padding: 25px;
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    border-radius: 5px;
    box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.15);
    background-color: var(--color-white);
    color: var(--color-twilight);
}

.text {
    font: var(--h4);
    font-weight: 600;
    margin: auto;
    margin-bottom: 0;
}

.resetButton {
    margin-top: 25px;
    cursor: pointer;
    user-select: none;
    max-width: 232px;
}

.buttonText {
    font-size: 18px;
    font-weight: 600;
    line-height: 45px;
}
