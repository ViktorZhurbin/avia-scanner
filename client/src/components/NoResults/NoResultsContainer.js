import { connect } from 'react-redux';

import NoResults from './NoResults';
import { resetStops } from '../../store/ticketData/actions';

const mapDispatchToProps = dispatch => ({
    resetFilter: () => dispatch(resetStops())
});

export default connect(
    null,
    mapDispatchToProps,
)(NoResults);
