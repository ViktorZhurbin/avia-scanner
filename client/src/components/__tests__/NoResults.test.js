import React from 'react';
import { shallow } from 'enzyme';

import NoResults from '../NoResults';

const props = {
    ticketsCount: 10,
    onFilterReset: () => null,
};

describe('Render NoResults', () => {
    it('render correctly', () => {
        const component = shallow(
            <NoResults {...props} />,
        );
        expect(component).toMatchSnapshot();
    });
});
