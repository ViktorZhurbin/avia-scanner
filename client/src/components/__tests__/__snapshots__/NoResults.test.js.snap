// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NoResults render correctly 1`] = `
<div
  className="container"
>
  <div
    className="text"
  >
    We found 10 tickets
            but none of them matches current filter settings
  </div>
  <div
    className="resetButton"
  >
    <Button
      isLoading={false}
      onClick={[Function]}
    >
      <span
        className="buttonText"
      >
        Reset Filters
      </span>
    </Button>
  </div>
</div>
`;
